src/generate/addResolveFunctionsToSchema.ts

```diff
@@ -51,6 +52,10 @@ function addResolveFunctionsToSchema(
     ? extendResolversFromInterfaces(schema, inputResolvers)
     : inputResolvers;
 
+  // Used to map the external value of an enum to its internal value, when
+  // that internal value is provided by a resolver.
+  const enumValueMap = {};
```

Comment:
Rather than using the object literal notation, which inherits from  `Object.prototype`, I believe we should be careful to use `Object.create(null)` here since the result of `getType(...).name` (a user-controlled string) could get set into it — potentially a problem with JavaScript internal properties like `length`, `prototype`, etc.

For example, without this (unless this is caught somewhere else that I'm missing?) weird behavior could ensue when the return value of `getType(...).name` is set into `enumValueMap` via `enumValueMap[valueReturnedFromGetTypeName] =`:

```js
  const sdl = `
    enum length { LONG SHORT }
    type Query { long: length }
  `;

  require("graphql")
    .buildSchema(sdl)
    .getType("length").name; // "length"
```

(And similarly with say, `constructor` or `prototype` in place of `length`).

Reply:
Oh yes, great point @abernix - I'll adjust, thanks!